Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,928 @@
+    {
+      var item = DropDownMenu.CreateItem(itemInfos[i], selectionCount);
+      if (item != null)
+        ul.appendChild(item);
+    }
+
+    var titleDivFunc = DropDownMenu.CreateTitleDivGetter (context);
+    DropDownMenu.ApplyPosition ($(div), evt, titleDivFunc());
+    $(div).iFrameShim({ top: '0px', left: '0px', width: '100%', height: '100%' });
+
+    if (DropDownMenu._repositionTimer) 
+      clearTimeout(DropDownMenu._repositionTimer);
+    var repositionHandler = function ()
+    {
+      if (DropDownMenu._repositionTimer)
+        clearTimeout (DropDownMenu._repositionTimer);
+
+      if (DropDownMenu._currentPopup && DropDownMenu._currentPopup == div && $(div).is (':visible'))
+      {
+        DropDownMenu.ApplyPosition ($(div), null, titleDivFunc());
+        DropDownMenu._repositionTimer = setTimeout (repositionHandler, DropDownMenu._repositionInterval);
+      }
+    };
+
+    // Only reposition if opened via titleDiv
+    if (evt == null)
+      DropDownMenu._repositionTimer = setTimeout(repositionHandler, DropDownMenu._repositionInterval);
+
+    if ($menuButton.data(DropDownMenu._button_timestampDataKey))
+    {
+      $menuButton.removeData(DropDownMenu._button_timestampDataKey);
+    }
+    else
+    {
+      DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+    }
+  }
+
+  private static CreateTitleDivGetter ($context: JQuery): () => JQuery
+  {
+    const context = $context[0];
+    const contextID = context.id;
+    if (contextID == null)
+    {
+      return function ()
+      {
+        // TODO: nullcheck?
+        return $(context.firstElementChild!);
+      };
+    }
+    else
+    {
+      return function ()
+      {
+        // TODO: shouldn't this be rewritten anyway? getElementById should only be called once here.
+        // TODO: nullability
+        return document.getElementById(contextID)! && $(document.getElementById(contextID)!.firstElementChild!);
+      };
+    }
+  }
+
+  private static ApplyPosition (popUpDiv: JQuery, clickEvent: Nullable<JQueryMouseEventObject>, referenceElement: JQuery): void
+  {
+    var space_top = Math.round(referenceElement.offset().top - $(document).scrollTop());
+    var space_bottom = Math.round($(window).height() - referenceElement.offset().top - referenceElement.height() + $(document).scrollTop());
+    var space_left = referenceElement.offset().left;
+    var space_right = $(window).width() - referenceElement.offset().left - referenceElement.width();
+
+    // position drop-down list
+    var top = clickEvent ? clickEvent.clientY : Math.max(0, space_top + referenceElement.outerHeight());
+    var left = clickEvent ? clickEvent.clientX : 'auto';
+    var right = clickEvent ? 'auto' : Math.max(0, $(window).width() - referenceElement.offset().left - referenceElement.outerWidth());
+
+    popUpDiv.css('top', top);
+    popUpDiv.css('bottom', 'auto');
+    popUpDiv.css('right', right as string);
+    popUpDiv.css('left', left as string);
```

Comment:
This is a bug since css values cannot just have values assigned except for `0` otherwise a unit (e.g. `px` is needed). Additionally we should not type cast here but convert to a string.

```suggestion
    // TODO: Missing unit
    popUpDiv.css('right', "" + right);
    popUpDiv.css('left', "" + left);
```